### PR TITLE
setup redux in the project

### PR DESCRIPTION
**Pull request by kennedyowusu**
## Highlights

- [x] Add redux toolkit 
- [x] Structure the app using feature folder
- [x] Write book actions and reducer using the ducks pattern
- [x] Write categories actions and reducer using the ducks pattern
- [x] Configure the redux store
- [x] Ensure all linters pass

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// This is a string constant that will be used to identify the action type
+const ADD_BOOK = 'bookStore/books/ADD_BOOK';
+const REMOVE_BOOK = 'bookStore/books/REMOVE_BOOK';
+
+const initialState = {
+  books: [],
+};
+
+// This is a function that returns an action object
+const addBook = createAction(ADD_BOOK);
```
Comment (Munsa1) [inline src/redux/books/books.js]: - [ ] Currently I am unable to add a book, kindly fix this issue

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+const removeBook = createAction(REMOVE_BOOK);
```
Comment (Munsa1) [inline src/redux/books/books.js]: - [ ] Even with the books that come already added to the app, I am unable to remove any book from the booklist. kindly fix this issue as well

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @kennedyowusu, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Passing Linters :heavy_check_mark: 
Correct GitHub Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (kennedyowusu) [reply src/redux/books/books.js]: This issue has been resolved

File: src/redux/books/books.js (hunk 2 above)
Comment (kennedyowusu) [reply src/redux/books/books.js]: This issue has been resolved

**Review by mikethreels — APPROVED** [review]
Hi Kennedy, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- linters are passing
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**